registry/tracecat_registry/core/http.py

```diff
@@ -22,12 +25,57 @@
 from tracecat.logger import logger
 from typing_extensions import Doc
 
+from tracecat.config import (
+    TRACECAT__MAX_FILE_SIZE_BYTES,
+    TRACECAT__MAX_AGGREGATE_UPLOAD_SIZE_BYTES,
+    TRACECAT__MAX_UPLOAD_FILES_COUNT,
+)
+
 from tracecat.types.exceptions import TracecatException
 from tracecat_registry import RegistrySecret, registry, secrets
 
 RequestMethods = Literal["GET", "POST", "PUT", "PATCH", "DELETE"]
 JSONObjectOrArray = dict[str, Any] | list[Any]
 
+
+class FileUploadData(TypedDict, total=False):
+    """Detailed file upload data structure."""
+
+    filename: str  # Filename sent in Content-Disposition header
+    content_base64: str  # Base64 encoded file content
+    content_type: str | None  # Optional MIME type (e.g., "image/png")
+
```

Comment:
prefer using typing.Required/typing.NotRequired instead of total=False if possible